src/doublycircularlist/DoublyCircularLinkedList.java

```diff
@@ -0,0 +1,186 @@
+package doublycircularlist;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.LinkedList;
+
+public class DoublyCircularLinkedList <E>{
+
+    private Node first;
+    private Node last;
+    public Integer size;
+
+    public DoublyCircularLinkedList(){
+        this.first = null;
+        this.last = null;
+        this.size = 0;
+    }
+
+
+    boolean add(E data) {
+
+        Node node = new Node(data);
+        if(this.size == 0){
+            this.first = node;
+            this.last = node;
+
+            first.setPrevious(last);
+            last.setNext(first);
+            size++;
+        }else{
+
+            node.setPrevious(last);
+            last.setNext(node);
+            node.setNext(first);
+            this.last = node;
+            size++;
+        }
+
```

Comment:
you can increase the size here instead of doing it two times.
The same goes for this.last=node;